frontend/packages/ceph-storage-plugin/src/components/volume-snapshot/volume-snapshot.tsx

```diff
@@ -0,0 +1,233 @@
+      props: { className: snapshotTableColumnClasses[5] },
+    },
+  ];
+};
+VolumeSnapshotTableHeader.displayName = 'SnapshotTableHeader';
+
+const kind = referenceFor(VolumeSnapshotModel);
+
+export const VolumeSnapshotDetails: React.SFC = (props) => {
+  const [resource, setResource] = React.useState(null);
+
+  React.useEffect(() => {
+    k8sGet(VolumeSnapshotModel, _.get(props, 'match.params.name'), _.get(props, 'match.params.ns'))
+      .then((volumeSnapshotObj) => {
+        try {
+          setResource(volumeSnapshotObj);
+        } catch (e) {
+          setResource(null);
+        }
+      })
+      .catch((error) => {
+        setResource(null);
+        throw error;
+      });
+  }, [props]);
+
+  return (
+    <div>
+      <div className="co-create-operand__header">
+        <div className="co-create-operand__header-buttons">
+          <BreadCrumbs
+            breadcrumbs={[
+              {
+                name: PersistentVolumeClaimModel.labelPlural,
+                path: `/k8s/ns/${_.get(props, 'match.params.ns')}/persistentvolumeclaims`,
+              },
+              {
+                name: `${_.get(props, 'match.params.appName')}`,
+                path: `/k8s/ns/${_.get(props, 'match.params.ns')}/persistentvolumeclaims/${_.get(
+                  props,
+                  'match.params.appName',
+                )}`,
+              },
+              {
+                name: 'Snapshot Details',
+                path: ``,
+              },
+            ]}
+          />
+        </div>
+
+        <h2 className="co-m-pane__heading co-m-pane__heading--baseline">
+          <div className="co-m-pane__name">
+            <ResourceIcon kind="VolumeSnapshot" /> {_.get(props, 'match.params.name')}
+          </div>
+        </h2>
+      </div>
+      <div className="co-m-pane__body">
+        <SectionHeading text="Snapshot Overview" />
+        <div className="row">
+          <div className="col-sm-6">
+            <dl>
+              <dt>Name</dt>
+              <dd>{_.get(resource, 'metadata.name')}</dd>
+              <dt>Date</dt>
+              <dd>{_.get(resource, 'metadata.creationTimestamp')}</dd>
+              <dt>Status</dt>
+              <dd>{_.get(resource, 'status.readyToUse') ? 'Ready' : 'Not Ready'}</dd>
+              <dt>Size</dt>
+              <dd>
+                {_.get(resource, 'status.restoreSize')
+                  ? _.get(resource, 'status.restoreSize')
+                  : 'No Data'}
+              </dd>
+              <dt>Namespace</dt>
+              <dd>
+                <ResourceLink kind="Namespace" />
+                {_.get(resource, 'metadata.namespace')}
+              </dd>
+              <dt>Annotations</dt>
+              <dd>None</dd>
+              <dt>Labels</dt>
+              <dd>None</dd>
+            </dl>
+          </div>
+          <div className="col-sm-6">
+            <dl>
+              <dt>API Version</dt>
+              <dd>{_.get(resource, 'apiVersion')}</dd>
```

Comment:
Why are we showing the api version? We don't show that for other resources.

Reply:
Attaching design doc: https://marvelapp.com/100ah4f4/screen/64180287
@yuvalgalanti can we keep it consistent with the other screens?